api/src/main/java/io/opentelemetry/trace/AttributeValue.java

```diff
@@ -136,6 +188,54 @@ public double getDoubleValue() {
         String.format("This type can only return %s data", getType().name()));
   }
 
+  /**
+   * Returns the String array value of this {@code AttributeValue}. An UnsupportedOperationException
+   * will be thrown if getType() is not {@link Type#STRING_ARRAY}.
+   *
+   * @return the array values of this {@code AttributeValue}.
+   * @since 0.3.0
+   */
+  public List<String> getStringArrayValue() {
+    throw new UnsupportedOperationException(
```

Comment:
Since API calls are not supposed to throw exceptions, should we instead just return an empty List for these cases? I'm not sure of what the right behavior is.

Reply:
Also basic types return an exception...
My proposal is to left as is. In a further follow up PR that addresses the renaming of the `AttributeValue` class ([context](https://github.com/open-telemetry/opentelemetry-java/pull/883)), we should rethink the APIs.
WDYT?

Reply:
Btw, these ones are meant to throw exceptions, by design - user is expected, upon getting an attr, to do a check like this:

```java
switch (attr.getType()) {
  case STRING:
    return attr.getStringValue();
  case LONG:
    return attr.getLongValue();
}
```

Reply:
as long as we're consistent, then 👍